source/common/router/config_impl.cc

```diff
@@ -523,14 +539,18 @@ const VirtualHostImpl* RouteMatcher::findVirtualHost(const Http::HeaderMap& head
     return default_virtual_host_.get();
   }
 
-  auto iter = virtual_hosts_.find(headers.Host()->value().c_str());
+  const char* host = headers.Host()->value().c_str();
+  auto iter = virtual_hosts_.find(host);
   if (iter != virtual_hosts_.end()) {
     return iter->second.get();
-  } else if (default_virtual_host_) {
-    return default_virtual_host_.get();
   }
-
-  return nullptr;
+  if (!wildcard_virtual_host_suffixes_.empty()) {
+    VirtualHostSharedPtr vhost = findWildcardVirtualHost(host);
```

Comment:
Somewhat complex dance required here because one of these methods returns a shared_ptr and one returns a bare pointer... why is it that way?